.github/workflows/release.yml

```diff
@@ -1,45 +1,34 @@
-name: Release to Hackage
+name: Release
 
 on:
   push:
-    tags:
-      - 'bcp47-v*'
-      - 'bcp47-orphans-v*'
-
-  workflow_dispatch:
-    inputs:
-      tag:
-        description: Tag to release
-        required: true
+    branches: main
 
 jobs:
   release:
+    runs-on: ubuntu-latest
+
     strategy:
       matrix:
         package:
           - bcp47
           - bcp47-orphans
-    runs-on: ubuntu-latest
-    env:
-      HACKAGE_API_KEY: ${{ secrets.HACKAGE_API_KEY }}
+
     steps:
       - uses: actions/checkout@v2
-      - id: gate
-        name: "Gate by package"
-        run: |
-          # Read tag from push event or manual trigger
-          event_tag=${{ github.event.inputs.tag }}
-          event_tag=${event_tag:-${{ github.ref }}}
-          job_package=${{ matrix.package }}
 
-          if [[ "$event_tag" == "$job_package-v"* ]]; then
-            echo "::set-output name=proceed::true"
-          else
-            echo "$event_tag does not match $job_package-v*"
-          fi
+      - id: tag
+        uses: freckle/haskell-tag-action@main
+        with:
+          working-directory: ${{ matrix.package }}
```

Comment:
This Action uses a different approach, since it only needs to identify a single file and working-directory doesn't matter:

```suggestion
          package-yaml: ${{ matrix.package }}/package.yaml
```